config/routes.rb

```diff
@@ -17,6 +17,8 @@
   end
 
   resources :nodes, only: [:index, :show, :destroy] do
+    post :assign
```

Comment:
The path can be `assign-roles` because as arguments we get the `master` the user choose, but we are going to set the minion role for the rest of the discovered machines. So you are effectively assigning roles (plural). The master that was chosen, and the rest, that will be minions by default (at least for now).